src/arrayserialize.rs

```diff
@@ -46,13 +52,17 @@ impl<'a, A, D> serde::ser::MapVisitor for AVisitor<'a, A, D>
     {
         match self.state {
             0 => {
-                self.state += 1;
-                Ok(Some(try!(serializer.visit_map_elt("shape", self.arr.dim()))))
-            }
+                self.state +=1;
+                Ok(Some(try!(serializer.serialize_map_elt("v", ARRAY_FORMAT_VERSION))))
+            },
             1 => {
                 self.state += 1;
-                Ok(Some(try!(serializer.visit_map_elt("data", self.arr.iter()))))
-            }
+                Ok(Some(try!(serializer.serialize_map_elt("dim", self.arr.dim()))))
+            },
+            2 => {
+                self.state += 1;
+                Ok(Some(try!(serializer.serialize_map_elt("data", self.arr.iter()))))
```

Comment:
Ok, it works as a drop-in replacement, the results are the same. Truth be told, I don't know what the semantically version would be. Pushed a commit for using the `serialize_struct_elt` version.